core/src/autogluon/core/utils/utils.py

```diff
@@ -273,26 +300,82 @@ def get_pred_from_proba(y_pred_proba, problem_type=BINARY):
-        stratify = None
-    else:
+    X_split = X
+    y_split = y
+    if problem_type in [BINARY, MULTICLASS]:
         stratify = y
+    else:
+        stratify = None
 
-    # TODO: Enable stratified split when y class would result in 0 samples in test.
-    #  One approach: extract low frequency classes from X/y, add back (1-test_size)% to X_train, y_train, rest to X_test
-    #  Essentially stratify the high frequency classes, random the low frequency (While ensuring at least 1 example stays for each low frequency in train!)
-    #  Alternatively, don't test low frequency at all, trust it to work in train set. Risky, but highest quality for predictions.
-    X_train, X_test, y_train, y_test = train_test_split(X, y.values, test_size=test_size, shuffle=True, random_state=random_state, stratify=stratify)
+    # This code block is necessary to avoid crashing when performing a stratified split and only 1 sample exists for a class.
+    # This code will ensure that the sample will be part of the train split, meaning the test split will have 0 samples of the rare class.
+    rare_indices = None
+    if stratify is not None:
+        cls_counts = y.value_counts()
+        cls_counts_invalid = cls_counts[cls_counts < min_cls_count_train]
+
+        if len(cls_counts_invalid) > 0:
+            logger.error(f'ERROR: Classes have too few samples to split the data! At least {min_cls_count_train} are required.')
+            logger.error(cls_counts_invalid)
+            raise AssertionError('Not enough data to split data into train and val without dropping classes!')
+        elif min_cls_count_train < 2:
+            cls_counts_rare = cls_counts[cls_counts < 2]
+            if len(cls_counts_rare) > 0:
+                cls_counts_rare_val = set(cls_counts_rare.index)
+                y_rare = y[y.isin(cls_counts_rare_val)]
+                rare_indices = list(y_rare.index)
+                X_split = X.drop(index=rare_indices)
+                y_split = y.drop(index=rare_indices)
+                stratify = y_split
+
+    X_train, X_test, y_train, y_test = train_test_split(X_split, y_split.values, test_size=test_size, shuffle=True, random_state=random_state, stratify=stratify)
     if problem_type != SOFTCLASS:
         y_train = pd.Series(y_train, index=X_train.index)
         y_test = pd.Series(y_test, index=X_test.index)
     else:
         y_train = pd.DataFrame(y_train, index=X_train.index)
         y_test = pd.DataFrame(y_test, index=X_test.index)
+
+    if rare_indices:
+        X_train = X_train.append(X.loc[rare_indices])
+        y_train = y_train.append(y.loc[rare_indices])
+
+    if problem_type in [BINARY, MULTICLASS]:
+        class_counts_dict_orig = y.value_counts().to_dict()
+        class_counts_dict = y_train.value_counts().to_dict()
+        class_counts_dict_test = y_test.value_counts().to_dict()
+
+        indices_to_move = []
+        random_state = random.getstate()
+        random.seed(random_state)
```

Comment:
Why is this needed? It looks like noop.

Reply:
Good catch! This was a bug, it is supposed to apply the random_state param and then reset to the outer random_state context.

Reply:
Fixed